src/components/StorageOverview/Capacity/Capacity.js

```diff
@@ -9,48 +11,131 @@ import {
   DashboardCardHeader,
   DashboardCardTitle,
 } from '../../Dashboard/DashboardCard';
+import { Dropdown } from '../../Form/Dropdown';
+import { DashboardCardActionsBody } from '../../Dashboard/DashboardCard/DashboardCardActionsBody';
+
 import { StorageOverviewContext } from '../StorageOverviewContext';
 import { CapacityItem } from '../../Dashboard/Capacity/CapacityItem';
 import { formatBytes } from '../../../utils';
 import { getCapacityStats } from '../../../selectors';
 import { CapacityBody } from '../../Dashboard/Capacity/CapacityBody';
+import { TOTAL, CAPACITY, VMSVSPODS } from './strings';
+
+const capacityTypes = [TOTAL, CAPACITY, VMSVSPODS];
+
+export const CapacityImplementation = ({ Total, Used, title, LoadingComponent }) => (
+  <CapacityBody>
+    <CapacityItem
+      id="capacity"
+      title={title}
+      used={getCapacityStats(Used)}
+      total={getCapacityStats(Total)}
+      formatValue={formatBytes}
+      LoadingComponent={LoadingComponent}
+      isLoading={!(Used && Total)}
+    />
+  </CapacityBody>
+);
 
 export class Capacity extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      capacity: capacityTypes[0],
+    };
+  }
+
+  convertMetrics = (Total, Used, title) => {
```

Comment:
so this function converts {Total, Used, title} to {metric1, metric2, title} and then it is passed back to Total/Used/title in https://github.com/kubevirt/web-ui-components/pull/475/files#diff-0be8d3e2808ab296537ca229a7f5a817R104 . Why use it at all ?

Reply:
I used this function to convert the different metrics used for `total` and `used` to a common variable so that it can be passed to the function to implement the Donut chart easily.
Example - For `Requested vs Used` , values for total, used and title are different when compared to values for `VMs vs Pods` .  https://github.com/kubevirt/web-ui-components/pull/475/files#diff-0be8d3e2808ab296537ca229a7f5a817R67
The variable names are repeated as I couldn't think of any other suitable name.

Reply:
ah, ok. But it could be slimmed down to 
```suggestion
  convertMetrics = (total, used, title) => ({total, used, title});
```